pallets/pool-registry/src/mock.rs

```diff
@@ -222,6 +223,7 @@ impl<
 		_currency: <T as pallet_pool_registry::Config>::CurrencyId,
 		_max_reserve: <T as pallet_pool_registry::Config>::Balance,
 	) -> DispatchResult {
+		#[cfg(feature = "runtime-benchmarks")]
```

Comment:
Are we sure the only code paths for this can be thru the runtime-benchmarks feature? I'm a little concerned about having the mocks behave differently between benches and tests...

Reply:
Good point! It's true that it's used in tests even when `runtime-benchmarks` is not enabled, so in that case, you're not creating a pool. Anyways, it is compiling and passing with and without `runtime-benchmarks` feature, so I suppose it should behave as expected.

Looking more deeply creating a pool seems to doesn't matter for the testing, but yes for the benchmarks because it adds a lot of weight to the call, so makes sense to only call `create_pool()` on benchmaking.